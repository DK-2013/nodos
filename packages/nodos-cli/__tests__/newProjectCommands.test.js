import fsp from 'fs/promises';
import path from 'path';
import os from 'os';
import { delay } from 'nanodelay';
import runNew from '../lib/new/index.js';

let dir;

beforeEach(async () => {
  dir = await fsp.mkdtemp(path.join(os.tmpdir(), 'nodos-'));
});

test('nodos/new', async () => {
  const appPath = 'site';
  const projectRoot = path.join(dir, appPath);
  const options = { exitProcess: false, args: ['new', appPath, '--skip-install'] };
  await runNew(dir, options);
  await delay(1000);
  const fileNames = await fsp.readdir(projectRoot);
  expect(fileNames).toMatchSnapshot();
});
