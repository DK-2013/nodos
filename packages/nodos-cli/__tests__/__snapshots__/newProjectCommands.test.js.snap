// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`nodos/new 1`] = `
Array [
  "Makefile",
  "app",
  "config",
  "package.json",
  "tests",
]
`;
